app/controllers/APIController.scala

```diff
@@ -28,10 +27,14 @@ class APIController @Inject() (
     status.asJson.spaces2
   }
 
-  def parseFromJson(request: Request[AnyContent]): Option[ProductFromJson] = {
+  def parseFromJson(
+      request: Request[AnyContent]
+  ): Option[ProductFromJson] = {
     val parseResult: Option[ProductFromJson] =
-      parser.decode[ProductFromJson](request.body.toString) match {
-        case error =>
+      parser.decode[ProductFromJson](
+        request.body.asInstanceOf[AnyContentAsJson].json.toString()
```

Comment:
you are doing the same thing as before, otherwise you wouldn't be able to access the json object directly by doing `.json`. I think it will be easiest if you read my comments above and explore using circe's body parsers